edx_exams/apps/api/v1/views.py

```diff
@@ -440,36 +444,50 @@ def get(self, request):
         """
         HTTP GET handler to fetch all exam attempt data
 
+        TODO: The endpoint should be refactored on deprecation of edx-proctoring.
+        Most of the complexity here has to do with the fact that we are supporting
+        slightly different API behaviors from that service.
+
         Parameters:
             None
 
         Returns:
             A Response object containing all `ExamAttempt` data.
         """
         user = request.user
-        latest_attempt = get_latest_attempt_for_user(user.id)
-        serialized_attempt = StudentAttemptSerializer(latest_attempt)
-
-        # if there is an active attempt in this service, return it
-        if latest_attempt is not None:
-            latest_attempt = check_if_exam_timed_out(latest_attempt)
-            serialized_attempt = StudentAttemptSerializer(latest_attempt)
-            if latest_attempt.status in (ExamAttemptStatus.started, ExamAttemptStatus.ready_to_submit):
+        exam_content_id = request.GET.get('content_id', None)
+        exam = get_exam_by_content_id(exam_content_id)
+
+        # if a specific exam is requested always return that exam
+        if exam is not None:
+            attempt = get_current_exam_attempt(user.id, exam.id)
+        else:
+            attempt = get_active_attempt_for_user(user.id)
+
+        serialized_attempt = StudentAttemptSerializer(attempt)
+
+        # if there is an active attempt in this service, return it.
+        if attempt is not None:
+            # An in progress attempt may be moved to 'submitted' if check_if_exam_timed_out
+            # determines that the attempt has timed out.
+            attempt = check_if_exam_timed_out(attempt)
+            serialized_attempt = StudentAttemptSerializer(attempt)
+            if attempt.status in (ExamAttemptStatus.started, ExamAttemptStatus.ready_to_submit):
                 return Response(status=status.HTTP_200_OK, data=serialized_attempt.data)
 
         # if edx-proctoring has an active attempt, return it
-        latest_attempt_legacy, response_status = get_active_exam_attempt(user.lms_user_id)
+        legacy_attempt, response_status = get_active_exam_attempt(user.lms_user_id)
         if (
-            latest_attempt_legacy is not None
-            and latest_attempt_legacy != {}
+            legacy_attempt is not None
+            and legacy_attempt != {}
             and response_status == status.HTTP_200_OK
         ):
-            return Response(status=status.HTTP_200_OK, data=latest_attempt_legacy)
+            return Response(status=status.HTTP_200_OK, data=legacy_attempt)
 
-        # otherwise return the latest attempt here regardless of status
+        # otherwise return the attempt from edx-exams regardless of status
         return Response(
             status=status.HTTP_200_OK,
-            data=serialized_attempt.data if latest_attempt is not None else {}
+            data=serialized_attempt.data if attempt is not None else {}
```

Comment:
Would it also be worth maybe checking if attempt is not None here, serializing the data, then sending it back in the response?